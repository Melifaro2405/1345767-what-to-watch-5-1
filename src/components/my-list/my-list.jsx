import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {fetchFavoriteFilmList} from "../../servi—Åes/api-actions";
import {filmProptypes} from "../../props-validation";
import {Link} from "react-router-dom";
import MoviesList from "../movies-list/movies-list";
import Footer from "../footer/footer";
import {AppRoute} from "../../consts";

class MyList extends PureComponent {

  componentDidMount() {
    const {loadFavoriteFilms} = this.props;
    loadFavoriteFilms();
  }

  render() {
    const {favoriteFilms, login} = this.props;
    const {avatar} = login;
    return (
      <div className="user-page">
        <header className="page-header user-page__head">
          <div className="logo">
            <Link to={AppRoute.ROOT} className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </Link>
          </div>

          <h1 className="page-title user-page__title">My list</h1>

          <div className="user-block">
            <div className="user-block__avatar">
              <img src={avatar} alt="User avatar" width="63" height="63"/>
            </div>
          </div>
        </header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>
          <MoviesList films={favoriteFilms} />
        </section>

        <Footer />
      </div>
    );
  }
}

MyList.propTypes = {
  favoriteFilms: PropTypes.arrayOf(PropTypes.shape(filmProptypes)).isRequired,
  loadFavoriteFilms: PropTypes.func.isRequired,
  login: PropTypes.shape({
    avatar: PropTypes.string
  })
};

const mapStateToProps = ({DATA, USER}) => ({
  favoriteFilms: DATA.favoriteFilms,
  login: USER.login
});

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteFilms() {
    dispatch(fetchFavoriteFilmList());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
