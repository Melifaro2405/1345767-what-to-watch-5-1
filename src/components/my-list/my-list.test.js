import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {MyList} from './my-list';

jest.mock(`../movies-list/movies-list`, () => `MoviesList`);
jest.mock(`../footer/footer`, () => `Footer`);

const noop = () => {};

export const films = [
  {
    id: 1,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 2,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 3,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  }
];

it(`Should MyList render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <MyList
            favoriteFilms={films}
            loadFavoriteFilms={noop}
            login={{avatar: `test`}}
          />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
