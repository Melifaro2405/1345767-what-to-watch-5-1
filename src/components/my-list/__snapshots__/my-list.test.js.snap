// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="test"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <MoviesList
      films={
        Array [
          Object {
            "details": Object {
              "runtime": 99,
            },
            "id": 1,
            "moreInfo": Object {
              "backgroundColor": "#ffffff",
              "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
              "genre": "Comedy",
              "isAddToMyList": false,
              "playPreviewSrc": "https://some-link",
              "playVideoSrc": "https://some-link",
              "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
              "releaseDate": 2014,
            },
            "overview": Object {
              "actorsList": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
              "director": "Wes Andreson",
              "rating": 8.9,
              "ratingCount": 240,
            },
            "preview": Object {
              "src": "img/the-grand-budapest-hotel-poster.jpg",
              "title": "The Grand Budapest Hotel",
            },
          },
          Object {
            "details": Object {
              "runtime": 99,
            },
            "id": 2,
            "moreInfo": Object {
              "backgroundColor": "#ffffff",
              "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
              "genre": "Comedy",
              "isAddToMyList": false,
              "playPreviewSrc": "https://some-link",
              "playVideoSrc": "https://some-link",
              "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
              "releaseDate": 2014,
            },
            "overview": Object {
              "actorsList": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
              "director": "Wes Andreson",
              "rating": 8.9,
              "ratingCount": 240,
            },
            "preview": Object {
              "src": "img/the-grand-budapest-hotel-poster.jpg",
              "title": "The Grand Budapest Hotel",
            },
          },
          Object {
            "details": Object {
              "runtime": 99,
            },
            "id": 3,
            "moreInfo": Object {
              "backgroundColor": "#ffffff",
              "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
              "genre": "Comedy",
              "isAddToMyList": false,
              "playPreviewSrc": "https://some-link",
              "playVideoSrc": "https://some-link",
              "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
              "releaseDate": 2014,
            },
            "overview": Object {
              "actorsList": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
              "director": "Wes Andreson",
              "rating": 8.9,
              "ratingCount": 240,
            },
            "preview": Object {
              "src": "img/the-grand-budapest-hotel-poster.jpg",
              "title": "The Grand Budapest Hotel",
            },
          },
        ]
      }
    />
  </section>
  <Footer />
</div>
`;
